scip/api/population.py

```diff
@@ -0,0 +1,80 @@
+from salmon_occurrence import ConservationUnit, Taxon, Reference, Population, Phenology
+from sqlalchemy_sqlschema import maintain_schema
+from sqlalchemy_sqlschema.sql import get_schema
+from sqlalchemy import func
+
+
+def population(
+    session,
+    overlap=None,
+    common_name=None,
+    scientific_name=None,
+    subgroup=None,
+    name=None,
+):
+    """Return information about salmon populations in the database that fulfill
+    specified parameters.
+
+    :param session: (sqlalchemy.orm.session.Session) a database Session object
+    :param overlap: a WKT string specifying a geometry that overlaps with the desired populations
+    :param common_name: common name of the salmon species of interest
+    :param scientific_name: scientific name of the salmon species of interest
+    :param subgroup: parameter designating a sub-species taxon, such as `lake` or `river` for sockeye salmon
+    :param name: name of the conservation unit that encloses the population's range
+
+    :return: a list of objects representing salmon populations that fulfill the given parameters. In
+        addition to `common_name`, `scientific_name`, `subgroup`, and `name`, two geoJSON strings describing
+        the geometry of the population7s extent are provided. `boundary` describes the outline of the extent,
+        and `outlet` provides the most downstream point of the extent according to the RVIC routed flow data.
+
+    """
+    # TODO: verify input arguments
+
+    # TODO: return additional data
```

Comment:
Interested to know what you mean by this!

Reply:
The original plan for this database included much more details information about salmon populations, such as their migration times. We have an extensive data set painstakingly gathered from a wide array of published sources - for the most part, each population of salmon is document in one or more separate scientific publications, and assembling the information was a challenge. 

Recently, scientists have voiced skepticism about some of the sources that informed the salmon dataset, so we've currently paused including that data in the website until we can verify things. 

You can see a schematic for the intended design of the dataset [here](https://github.com/pacificclimate/scip-db/blob/init/diagrams/v1.png), though I imagine it will look familiar to you, as you helped design it. :) Currently all those tables are present in the database, but empty and ignore by the API.

Codewise, the current API only returns the "basic" attributes - salmon species and subgroup, and does not return any of the unverified timing information. I am not sure if I should be returning a bunch of blank attributes for data that isn't in the database yet, or if I should wait until we see how much of that data is useable - I could imagine maybe we'd classify salmon populations by migration season, but not include exact weeks, or something else like that.